python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -796,6 +794,28 @@ def _get_job_def_for_asset_selection(
                 " or job."
             )
 
+        # Test that selected asset checks can be run individually. Currently this is only supported
+        # on checks defined with @asset_check, which will have an AssetChecksDefinition.
+        all_check_handles_in_checks_defs = set()
+        for asset_checks_def in self.asset_layer.asset_checks_defs:
+            for spec in asset_checks_def.specs:
+                all_check_handles_in_checks_defs.add(spec.handle)
+
+        non_checks_defs_asset_check = [
+            asset_check
+            for asset_check in asset_check_selection or set()
+            if asset_check not in all_check_handles_in_checks_defs
+        ]
+        non_checks_defs_asset_check = [
+            str(asset_check) for asset_check in non_checks_defs_asset_check
+        ]
+        if non_checks_defs_asset_check:
+            raise DagsterInvalidSubsetError(
+                "Asset checks provided in asset_check_selection argument"
+                f" {', '.join(non_checks_defs_asset_check)} are not defined with @asset_check and"
```

Comment:
Probably doesn't make sense to reference `@asset_check` here since I assume a user could define this with `AssetsCheckDefinition`?

```suggestion
                f" {', '.join(non_checks_defs_asset_check)} are not defined with @asset_check and"
```